pepper-apis/scripts/admin/delete-participant-data-for-study.sql

```diff
@@ -0,0 +1,183 @@
+-- scripts that will:
+-- generate a bunch of sql statements and shell script calls to be run to erase data for a study from dsm and dss.
+-- it also sets queued events to happen in the infinite future, effectively disabling them
+
+-- set the study to be deleted
+set @study_guid = 'singular' collate 'utf8mb4_unicode_ci';
+
+-- set the gcp environment in which to run the user deletes for dss
+set @gcp_project = 'broad-ddp-test';
+set @delete_sleep_seconds = 5;
+use peppertest;
+
+-- remove age up jobs
+delete from study_governance_policy
+where study_id in (select s.umbrella_study_id from umbrella_study s where s.guid = @study_guid)
+
+
+-- reset queued events to the future
+update queued_event set post_after = 99999999999999 where event_configuration_id in
+                                                          (select c.event_configuration_id
+                                                           from umbrella_study s,
+                                                                event_configuration c
+                                                           where s.guid = @study
+                                                             and s.umbrella_study_id = c.umbrella_study_id)
+                                                      and post_after > unix_timestamp();
+
+-- delete governed users in dss
+select distinct
+    concat('gcloud --project=', @gcp_project,  ' pubsub topics publish dss-tasks --attribute ''taskType=USER_DELETE,participantGuid=',
+           u.guid,',operatorGuid=andrew'' --message ''{"comment" : "purging singular data"} '';  sleep ',@delete_sleep_seconds, ';')
+from user u,
+     user_study_enrollment e,
+     umbrella_study s,
+     user_governance g
+where
+        u.user_id = e.user_id
+  and
+        e.study_id = s.umbrella_study_id
+  and
+        g.participant_user_id = u.user_id
+  and
+        s.guid = @study_guid;
+
+-- delete non governed users in dss
+select distinct
+    concat('gcloud --project=', @gcp_project,  ' pubsub topics publish dss-tasks --attribute ''taskType=USER_DELETE,participantGuid=',
+           u.guid,',operatorGuid=andrew'' --message ''{"comment" : "purging singular data"} ''; sleep ',@delete_sleep_seconds,';')
+from user u,
+     user_study_enrollment e,
+     umbrella_study s
+where
+        u.user_id = e.user_id
+  and
+        e.study_id = s.umbrella_study_id
+  and
+  -- don't include governed users
+    not exists (select 1 from user_governance g where g.operator_user_id = u.user_id)
+  and
+        s.guid = @study_guid;
+
+
+use test_dsm_db;
+
+-- delete kits
+select
+    concat('delete from ddp_kit where dsm_kit_id = ',k.dsm_kit_id, ';')
+from
+    ddp_kit_request req,
+    ddp_instance i,
+    ddp_kit k
+where
+        i.study_guid = @study_guid
```

Comment:
Probably need to confirm the study name (not really a GUI) on each environment, since these are not always consistent across environments.

Reply:
Added a comment to remind us to do this.